### DISCUSSION TITLE
Introducing TinyFlux: The time-series version of TinyDB

### DISCUSSION THREAD

**Discussion by citrusvanilla** [msiemens/tinydb#481] [General]
Hey TinyDB users- I am happy to announce that a version of TinyDB now exists for time-series data, called TinyFlux, available at https://github.com/citrusvanilla/tinyflux.

The work Markus and the TinyDB community has performed on TinyDB is admirable- I don't think I've ever read Python as clean and elegant and well-documented as what I found in the TinyDB codebase.  I was so inspired that I had to try to make TinyDB work for time-series data and the result is TinyFlux.

TinyFlux is very similar to TinyDB- it has basically the same query language but is extended for time-series data.  TinyFlux is backed by CSV however, so that writes may be append-only.  TinyFlux also creates an index in memory for efficient retrieval of the data, which is not currently a feature of TinyDB.  I envision TinyFlux as a storage layer for at-home IOT devices communicating with over MQTT, and as a single-user store of any time series data.

Make no mistake, TinyFlux is not as mature as TinyDB as I had to cut some features.  If you either need a time-series version of TinyDB, or want to contribute to a new-ish project, come check out TinyFlux.

Also, [here is a Medium article](https://citrusvanilla.medium.com/introducing-tinyflux-the-tiny-time-series-database-for-python-based-iot-analytics-applications-c3ef3c3bedf) introducing the project.  Give it a read!

Thanks for TinyDB, and happy coding!

~Justin

**Comment by msiemens** [comment]
Thanks for your kind words @citrusvanilla! And congratulations on releasing TinyFlux! 🙂
